eth/handler.go

```diff
@@ -410,8 +410,12 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 			bytes   common.StorageSize
 			headers []*types.Header
 			unknown bool
+			lookups int
 		)
-		for !unknown && len(headers) < int(query.Amount) && bytes < softResponseLimit && len(headers) < downloader.MaxHeaderFetch {
+		for !unknown && len(headers) < int(query.Amount) && bytes < softResponseLimit &&
+			len(headers) < downloader.MaxHeaderFetch &&
+			lookups < 2*downloader.MaxHeaderFetch {
+			lookups++
```

Comment:
Is the `lookups` always equal to `len(headers)`?

Reply:
Yes, I suppose it will be in this particular clause